crates/holochain/src/core/workflow/app_validation_workflow.rs

```diff
@@ -905,3 +964,16 @@ pub fn put_integration_limbo(
     set_validation_stage(txn, hash, ValidationLimboStatus::AwaitingIntegration)?;
     Ok(())
 }
+
+pub fn put_integrated(
+    txn: &mut Transaction<'_>,
+    hash: DhtOpHash,
+    status: ValidationStatus,
+) -> WorkflowResult<()> {
+    set_validation_status(txn, hash.clone(), status)?;
+    // This set the validation stage to pending which is correct when
+    // it's integrated.
+    set_validation_stage(txn, hash.clone(), ValidationLimboStatus::Pending)?;
+    set_when_integrated(txn, hash, Timestamp::now())?;
+    Ok(())
+}
```

Comment:
This runs in a tight loop for every integrated op, would it be worth doing this as a single update instead of three?

Reply:
It's within the same transaction and doesn't seem to be a bottleneck atm